src/Chat.tsx

```diff
@@ -74,6 +75,8 @@ export class Chat extends React.Component<ChatProps, {}> {
 
         konsole.log("BotChat.Chat props", props);
 
+        this.handleKeyDown = this.handleKeyDown.bind(this);
```

Comment:
Can this be bound to a React component? Doesn't it need a DOM element?

Reply:
We should do `.bind()` in constructor instead of inside `render()` because React do reference equals on props to find out if it need to be updated. If we do `.bind()` in `render()` loop, it'd result in a wasted render (i.e. a re-render that render nothing new, `React.addons.Perf.printWasted()`).